tasks/ftp/retrieve_from_ftp.py

```diff
@@ -0,0 +1,85 @@
+import os
+from ftplib import FTP, error_perm
+
+from netunicorn.base import Failure, Result, Success, Task
+
+
+class RetrieveFromFTP(Task):
+    """
+    Task for retrieving a file from an FTP server to a local directory. Establishes a connection to the specified FTP server, navigates to the desired remote directory and downloads the specified file to a local directory.
+    """
+
+    def __init__(
+        self,
+        ftp_remote_filepath: str,
+        ftp_url: str,
+        username: str,
+        password: str,
+        local_dir: str = "./",
+        *args,
+        **kwargs,
+    ):
+        """
+        Initializes the RetrieveFromFTP task with parameters.
+
+        Parameters:
+          ftp_remote_filepath (str): Full path of the file on the FTP server to retrieve.
+          ftp_url (str): URL or IP address of the FTP server.
+          username (str): Username for FTP authentication.
+          password (str): Password for FTP authentication.
+          local_dir (str, optional): Local directory to save the retrieved file. Defaults to "./".
+          *args: Variable length argument list.
+          **kwargs: Arbitrary keyword arguments.
+        """
+        super().__init__(*args, **kwargs)
+        self.ftp_remote_filepath = ftp_remote_filepath
+        self.ftp_url = ftp_url
+        self.username = username
+        self.password = password
+        self.local_dir = local_dir
+
+    def run(self) -> Result:
+        """
+        Run the FTP file retrieval process.
+
+        Steps:
+        1. Connects to FTP server using provided credentials.
+        2. Navigates to the directory containing the target file.
+        3. Downloads the specified file to the local directory. Defaults to "./".
+
+        Returns:
+          Result:
+            Success: Contain success message upon successful download.
+            --OR--
+            Failure: Contains an error message if the download fails.
+
+        """
+        try:
+            ftp = FTP(self.ftp_url, timeout=30)  # Modify timeout as needed
+            ftp.login(user=self.username, passwd=self.password)
+
+            remote_dir, remote_filename = os.path.split(self.ftp_remote_filepath)
+            if remote_dir:
+                ftp.cwd(remote_dir)
+
+            if not os.path.isdir(self.local_dir):
+                return Failure(f"Local directory does not exist: {self.local_dir}")
```

Comment:
I'd say almost always I'd prefer the task to create a folder if it does not exist.
Please, change returning Failure to folder creation using `os.makedirs`

Reply:
Also, please move this logic earlier to the task (before establishing the connection) to fail earlier if something locally is incorrect.